Packs/GooglePubSub/Integrations/GooglePubSub/GooglePubSub.py

```diff
@@ -1406,11 +1404,9 @@ def try_pull_unique_messages(
                 demisto.debug(
                     f"GCP_PUBSUB_MSG Duplicates with max_publish_time: {max_publish_time}"
                 )
-                return try_pull_unique_messages(
-                    client, sub_name, previous_msg_ids, retry_times - 1
-                )
-            # clean non-unique ids from raw_msgs
-            else:
+                return try_pull_unique_messages(client, sub_name, previous_msg_ids, retry_times - 1, ack_incidents)
```

Comment:
I would add a log msg here saying calling a recursive method with the amount of retries.
(As recursive methods are hard to debug perhaps logging will help in the future)